### PR TITLE
Add custom script support #46

### PR DESCRIPTION
**Pull request by ketan**

### CODE REVIEW

**Comment by ketan** [comment]
@varshavaradarajan — if you're on board with this change, I can apply custom script support to the docker-gocd-agent repo as well.

File: README.md.erb [hunk 1]
```diff
@@ -48,12 +53,52 @@ docker run -v /path/to/godata:/godata -v /path/to/home-dir:/home/go gocd/gocd-se
 
 All plugins can be installed under `/godata`.
 
+### Installing plugins using an environment configuration
+
+To install plugins, just add an ENV variable with the prefix `GOCD_PLUING_INSTALL_` and your name as `suffix`
```
Comment (varshavaradarajan) [inline README.md.erb]: PLUING => PLUGIN

File: install-gocd-plugins [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+yell() { echo "$0: $*" >&2; }
+die() { yell "$*"; exit 111; }
+try() { echo "$ $@" 1>&2; "$@" || die "cannot $*"; }
+
+# do that aslong we are not sure this is really set in the base image
+GOCD_PLUGIN_EXTERNAL_HOME=${GOCD_PLUGIN_EXTERNAL_HOME:-/godata/plugins/external}
```
Comment (varshavaradarajan) [inline install-gocd-plugins]: GOCD_PLUGIN_EXTERNAL_HOME - Can you please explain the comment about why we provide this when the go server will always look in /godata/plugins/external ? From the SystemEnvironment, I see that the external path is set as a system property.

File: Dockerfile [hunk 3]
```diff
@@ -47,6 +47,7 @@ RUN \
   mv go-server-17.11.0 /go-server
 
 COPY logback-include.xml /go-server/config/logback-include.xml
+COPY install-gocd-plugins /usr/local/sbin/install-gocd-plugins
```
Comment (varshavaradarajan) [inline Dockerfile]: This should be in Dockerfile.erb

**Review by varshavaradarajan — CHANGES_REQUESTED** [review]

File: install-gocd-plugins [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+GOCD_PLUGIN_EXTERNAL_HOME="/godata/plugins/external"
+
+if [ ! -e "${GOCD_PLUGIN_EXTERNAL_HOME}" ]; then
+  try mkdir -p "${GOCD_PLUGIN_EXTERNAL_HOME}"
+  try chown go:go "${GOCD_PLUGIN_EXTERNAL_HOME}"
+fi
+
+# List all vars beginning with `GOCD_PLUGIN_INSTALL_`
+PLUGIN_ENV_NAMES=( "$(compgen -A variable GOCD_PLUGIN_INSTALL_)" )
```
Comment (varshavaradarajan) [inline install-gocd-plugins]: compgen returns empty when I do `docker run -e GOCD_PLUGIN_INSTALL_github-pr=...`

**Comment by ketan** [comment]
Which OS and bash version is this?

**Comment by varshavaradarajan** [comment]
Alpine 3.6, Bash 4.3.48-r1

File: install-gocd-plugins [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  try mkdir -p "${GOCD_PLUGIN_EXTERNAL_HOME}"
+  try chown go:go "${GOCD_PLUGIN_EXTERNAL_HOME}"
+fi
+
+while IFS='=' read -r name value ; do
+  if [[ $name == *"GOCD_PLUGIN_INSTALL_"* ]]; then
+    plugin_url="${value}"
+
+    # remove the `GOCD_PLUGIN_INSTALL_` prefix and grab the plugin name
+    plugin_name="${env_var//GOCD_PLUGIN_INSTALL_/}"
```
Comment (varshavaradarajan) [inline install-gocd-plugins]: env_var => name

**Review by varshavaradarajan — CHANGES_REQUESTED** [review]

**Pull request merged**